ibis/expr/datatypes.py

```diff
@@ -602,6 +603,67 @@ def __str__(self) -> str:
         )
 
 
+class GeoSpatial(DataType):
+    __slots__ = 'geotype', 'srid'
+
+    def __init__(
+        self,
+        geotype: str = None,
+        srid: int = None,
+        nullable: bool = True
+    ):
+        """Geospatial data type base class
+
+        Parameters
+        ----------
+        geotype : str
+            Specification of geospatial type which could be `geography` or
+            `geometry`.
+        srid : int
+            Spatial Reference System Identifier
+        nullable : bool, optional
+            Whether the struct can be null
+        """
+        super().__init__(nullable=nullable)
+
+        if geotype not in (None, 'geometry', 'geography'):
+            raise ValueError(
+                'The `geotype` parameter should be `geometry` or `geography`'
+            )
+
+        self.geotype = geotype
+        self.srid = srid
+
+
+class Point(GeoSpatial):
+    """A point described by two coordinates."""
+    scalar = ir.PointScalar
+    column = ir.PointColumn
+
+    __slots__ = 'geotype', 'srid'
```

Comment:
@xmnlab These slots are already defined in `GeoSpatial`, but to prevent initializing `__dict__` You still need to define `__slot__ = ()` for each descendant classes like LineString below.

Reply:
OK sounds good I am doing that now .. thanks :+1: